paasta_tools/kubernetes_tools.py

```diff
@@ -1855,20 +1853,26 @@ def sanitize_for_config_hash(
         :param config: complete_config hash to sanitise
         :returns: sanitised copy of complete_config hash
         """
-        ahash = {
-            key: copy.deepcopy(value)
-            for key, value in config.to_dict().items()
-            if key not in CONFIG_HASH_BLACKLIST
-        }
-        spec = ahash["spec"]
-        ahash["spec"] = {
-            key: copy.deepcopy(value)
-            for key, value in spec.items()
-            if key not in CONFIG_HASH_BLACKLIST
-        }
+        ahash = config.to_dict()  # deep convert to dict
```

Comment:
I think I see at least [one case](https://github.com/Yelp/paasta/blob/7eba1996de7a51a016536e2ee4b2f3c9de413c09/paasta_tools/paastaapi/model_utils.py#L1495) where model_to_dict returns a value directly without copying. (i.e. if the model has a list of lists, the outer list will be copied, but the inner lists won't be, so if someone modifies the return value of this, we might actually modify the model object)

It looks like we only ever call this in one place (also in kubernetes_tools.py), where we immediately pass the result to `get_config_hash`, which just converts to JSON and then md5sums the result. I think my concern is purely theoretical. It would make for some spooky failure modes, though.